### PR TITLE
completed backend for viewing application details

### PR DESCRIPTION
**Pull request by jordantzj**

### CODE REVIEW

**Comment by renjie-teo** [comment]
Please test with Postman API and comment if all test cases pass automatically, with 'tests' configured with each request. Please also commit test data to be used for testing.

Ping me once the code review changes has been done, also do document API documentation for frontend team and attach screenshot of all collection test passes to Jira once complete, thank you.

**Review by renjie-teo — DISMISSED** [review]

File: Backend/apply/apply.py [hunk 1]
```diff
@@ -1,14 +1,29 @@
...
     applications = Apply.query.all()
     return jsonify([app.json() for app in applications])
 
-# for config testing
-@apply_routes.route('/test/configtest')
-def testConfigTest():
-    return current_app.config["DB_HOST"] 
\ No newline at end of file
+#get applications for a specfic role
+@apply_routes.route('/api/v1/viewApplicants/role/<int:id>')
```
[the remark this answers is not in the document]
Comment (jordantzj) [reply Backend/apply/apply.py:13]: no, handled my axios at the frontend

File: Backend/models.py [hunk 2]
```diff
@@ -25,3 +25,51 @@ def __init__(self, application_id, applicant_staff_id, applicant_existing_role,
     def json(self):
         return {"application_id": self.application_id, "applicant_staff_id": self.applicant_staff_id, "applicant_existing_role": self.applicant_existing_role, "applicant_existing_dept": self.applicant_existing_dept, "application_status": self.application_status, "date_applied": self.date_applied, "applied_role_id": self.applied_role_id}
 
+class ApplySkill(db.Model):
```
[the remark this answers is not in the document]
Comment (jordantzj) [reply Backend/models.py:28]: will update class diagram

File: Backend/models.py [hunk 3]
```diff
@@ -25,3 +25,51 @@ def __init__(self, application_id, applicant_staff_id, applicant_existing_role,
...
+    applicant_skills = db.Column(db.String(50), nullable=False)
+
+    def __init__(self, application_id, applicant_skills):
+        self.application_id = application_id
+        self.applicant_skills = applicant_skills
+
+    def json(self):
+        return {"application_id": self.application_id, "application_skills": self.applicant_skills}
+    
+class Role(db.Model):
```
[the remark this answers is not in the document]
Comment (jordantzj) [reply Backend/models.py:40]: no

File: Backend/models.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
 from extensions import db
 
 class Apply(db.Model):
```
[the remark this answers is not in the document]
Comment (jordantzj) [reply Backend/models.py:6]: no

File: Backend/models.py [hunk 5]
```diff
@@ -25,3 +25,51 @@ def __init__(self, application_id, applicant_staff_id, applicant_existing_role,
...
+        self.department = department
+        self.job_description = job_description
+        self.original_creation_dt = original_creation_dt
+        self.expiry_dt = expiry_dt
+        self.hiring_manager_id = hiring_manager_id
+
+    def json(self):
+        return {"role_id": self.role_id, "role_name": self.role_name, "job_type": self.job_type, "department": self.department, "job_description": self.job_description, "original_creation_dt": self.original_creation_dt, "expiry_dt": self.expiry_dt, "hiring_manager_id": self.hiring_manager_id}
+    
+class RoleSkill(db.Model):
```
[the remark this answers is not in the document]
Comment (jordantzj) [reply Backend/models.py:64]: no

File: Backend/role/role.py [hunk 6]
```diff
@@ -1,15 +1,16 @@
...
 
 role_routes = Blueprint('role_routes', __name__)
 
-# for admin
-@role_routes.route('/viewRoles')
-def viewApplicants():
-    return 'View Roles'
+#get all roles
+@role_routes.route('/api/v1/viewRoles')
+def viewRoles():
```
[the remark this answers is not in the document]
Comment (jordantzj) [reply Backend/role/role.py:8]: no, done in frontend

File: Backend/role/role.py [hunk 7]
```diff
@@ -1,15 +1,16 @@
...
+    roles = Role.query.all()
+    return jsonify([role.json() for role in roles])
 
-# for config testing
-@role_routes.route('/test/configtest')
-def testConfigTest():
-    return current_app.config["DB_HOST"] 
\ No newline at end of file
+#get skills needed for a specific role
+@role_routes.route('/api/v1/viewRoles/skill/<string:name>')
```
[the remark this answers is not in the document]
Comment (jordantzj) [reply Backend/role/role.py:13]: no, done in frontend

**Review by renjie-teo — APPROVED** [review]

**Pull request merged**